ebos/eclgenerictracermodel.cc

```diff
@@ -243,6 +243,43 @@ linearSolve_(const TracerMatrix& M, TracerVector& x, TracerVector& b)
     return result.converged;
 }
 
+template<class Grid,class GridView, class DofMapper, class Stencil, class Scalar>
+bool  EclGenericTracerModel<Grid,GridView,DofMapper,Stencil,Scalar>::
+linearSolveBatchwise_(const TracerMatrix& M, std::vector<TracerVector>& x, std::vector<TracerVector>& b)
+{
+#if ! DUNE_VERSION_NEWER(DUNE_COMMON, 2,7)
+    Dune::FMatrixPrecision<Scalar>::set_singular_limit(1.e-30);
+    Dune::FMatrixPrecision<Scalar>::set_absolute_limit(1.e-30);
+#endif
+    Scalar tolerance = 1e-2;
+    int maxIter = 100;
+
+    int verbosity = 0;
+    using TracerSolver = Dune::BiCGSTABSolver<TracerVector>;
+    using TracerOperator = Dune::MatrixAdapter<TracerMatrix,TracerVector,TracerVector>;
+    using TracerScalarProduct = Dune::SeqScalarProduct<TracerVector>;
+    using TracerPreconditioner = Dune::SeqILU< TracerMatrix,TracerVector,TracerVector>;
+
+    TracerOperator tracerOperator(M);
+    TracerScalarProduct tracerScalarProduct;
+    TracerPreconditioner tracerPreconditioner(M, 0, 1); // results in ILU0
+
+    TracerSolver solver (tracerOperator, tracerScalarProduct,
+                         tracerPreconditioner, tolerance, maxIter,
+                         verbosity);
+
+    bool converged = true;
+    for (int nrhs =0; nrhs < b.size(); ++nrhs) {
+        x[nrhs] = 0.0;
+        Dune::InverseOperatorResult result;
+        solver.apply(x[nrhs], b[nrhs], result);
+        converged = (converged && result.converged);
```

Comment:
@blattms Can this be solved more efficiently using multiple right-hand-sides directly. i.e. utilizing vectorization.

Reply:
Thanks for your for comments. Then we agree that:
1. Then the Dual-stuff will be removed.
2. Will add some comments in the code.

Multiple rhs:  Currently we use std::vector as containers for the vectors. Migrating to a dune container is definitively on the agenda, but perhaps not for this PR?

Reply:
> Then we agree that:
    1. Then the Dual-stuff will be removed.
    2. Will add some comments in the code.

Yes. I like the way you are planning to include the Dual-stuff. But lets do that in a separate PR later. 

> Multiple rhs: Currently we use std::vector as containers for the vectors. Migrating to a dune container is definitively on the agenda, but perhaps not for this PR? 

The current approach is fine for this PR.